pages/login.page.ts

```diff
@@ -0,0 +1,27 @@
+import { Locator, Page } from "@playwright/test";
+
+export class LoginPage {
+    readonly loginInput: Locator;
+    readonly passwordInput: Locator;
+    readonly loginButton: Locator;
+    readonly errorMsgHeader: Locator;
+    readonly errorMsgParagraph: Locator;
+
+    /* Notatka-1: Dlaczego wymagany jest private dla page. 
+    Ew. Dlaczego np. w dokumentacji PW w konstruktorze 
+    inicjalizują page poprzez this.page = page */
```

Comment:
Ogólnie ja nie lubię classowego podejścia. Wprowadza sporo zamieszania. Wolę normalne funkcje zwłaszcza, że classy w jsie to funckje... 

Moja propozycja na to:

```
import { Page } from '@playwright/test';

function getElements(page: Page) {
  return {
    loginInput: page.locator('#email'),
    passwordInput: page.locator('#passwd'),
    loginButton: page.locator('#SubmitLogin'),
    errorMsgHeader: page.locator('div.alert-danger p'),
    errorMsgParagraph: page.locator('div.alert-danger li'),
  };
}

export function LoginPage(page: Page) {
  const $el = getElements(page);

  return {
    $el,

    async singInUser(userEmail: string, password: string): Promise<void> {
      await $el.loginInput.fill(userEmail);
      await $el.passwordInput.fill(password);
      await $el.loginButton.click();
    },
  };
}
```

Reply:
Never thought that we can return object.